superset/databases/api.py

```diff
@@ -607,6 +612,11 @@ def test_connection(  # pylint: disable=too-many-return-statements
             TestConnectionDatabaseCommand(g.user, item).run()
             return self.response(200, message="OK")
         except DatabaseTestConnectionFailedError as ex:
+            engine = item["sqlalchemy_uri"].split("://")[0]
+            self.incr_stats(
+                f"db_test_connection_failed.{engine}.{ex.__class__.__name__}",
```

Comment:
`ex.__class__.__name__` will always be recorded as `DatabaseTestConnectionFailedError` if I understand this correctly. To capture the exception surfaced by the engine, you'd need to alter the command to surface that class name as part of the error object or move this logging into the command.